### PR TITLE
WIP: Upgrade to gradle android build tool 3.0

### PR DESCRIPTION
**Pull request by adikari**

### CODE REVIEW

**Comment by adikari** [comment]
Travis seems to be failing due to license acceptance. I will update travis config if this PR gets interest of author.

File: buildsystem/dependencies.gradle [hunk 1]
```diff
@@ -6,10 +6,10 @@ allprojects {
 
 ext {
   //Android
-  androidBuildToolsVersion = "23.0.1"
+  androidBuildToolsVersion = "26.0.2"
   androidMinSdkVersion = 15
-  androidTargetSdkVersion = 21
-  androidCompileSdkVersion = 21
+  androidTargetSdkVersion = 25
+  androidCompileSdkVersion = 25
```
Comment (piotrek1543) [inline buildsystem/dependencies.gradle]: I would change them to 26 to be more consistent. 

I think that there were used `21`(much lower to build tools version) just to do not need to implement Android permission runtime for release version of app.

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,6 @@
-#Tue Jan 19 23:59:31 CET 2016
+#Thu Oct 26 20:34:00 AEDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-2.10-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-4.2.1-bin.zip
```
Comment (piotrek1543) [inline gradle/wrapper/gradle-wrapper.properties]: there is already `4.3`

**Comment by piotrek1543** [comment]
@adikari you changed tools version in Gradle, but forgot to change it in CI config (which is a common mistake in PR like this). That's why building project fails.

**Pull request closed without merging**